### ISSUE TITLE
menu: invalid URL when the shortcut is an internal link

### ISSUE THREAD

**Issue by garz75** [McShelby/hugo-theme-relearn#163]
Hello and thank you for this fork.

We recently switched from learn to relearn to address issues not fixed in learn. However, we have now a small issue with relearn regarding the way it manages differently the "internal links" in the extra shortcuts menu, compared to learn:

My configuration is the following: 
```
baseURL: "https://docs.mysite.com/current"
relativeURLs: false
menu:
  shortcuts:
    - name: Server
      url: /server/openapi.html
```

When using the learn theme, the generated HTML is the following: 
```
              <li>
                  <a class="padding" href="https://docs.site.com/current/server/openapi.html">Server</a>
              </li>
```
When using relearn, we get: 

```
            <li><a class="padding" href="/current/current/server/openapi.html">Server</a></li>
```
which is obviously wrong...
I am not an expert at all with hugo and templates, the only notable difference I see in the partial for  the menus is: 
```
$ ag '\.URL' themes/*/layouts/partials/menu.html
themes/learn/layouts/partials/menu.html
35:                  {{.Pre}}<a class="padding" href="{{.URL | absLangURL }}">{{safeHTML .Name}}</a>{{.Post}}

themes/relearn/layouts/partials/menu.html
48:            <li>{{ .Pre }}<a class="padding" href="{{ .URL | relLangURL }}">{{ safeHTML .Name }}</a>{{ .Post }}</li>
``` 

relearn uses `relLangURL` instead of `absLangURL` to generate the full URL for the menu link...

I temporarily fixed the issue by changing the configuration: 
```
baseURL: "https://docs.site.com/current"
relativeURLs: false
menu:
  shortcuts:
    - name: Server
      url: https://docs.site.com/current/server/openapi.html
```

But this is not great because we have several versions of the doc (pre-prod/prod), with different values for the baseURL, it is not great to have to maintain 2 different configuration files for the menus for different environments...

Is there a way to work around this issue ? 

Thank you in advance

**Comment by McShelby** [comment]
I agree that your workaround is just, well, a workaround ;-)

To further tackle this down, could you attach your config.toml to this issue and the commandline you are using to run Hugo? This may be related to your @baseURL@ setting is my first guess.

**Comment by McShelby** [comment]
Is `/server/openapi.html` created by Hugo or is it some static page? Could you try `/server/openapi/` instead?

**Comment by garz75** [comment]
> Is `/server/openapi.html` created by Hugo or is it some static page? Could you try `/server/openapi/` instead?

It is a static page, generated elsewhere (via redoc). Using `/server/openapi/` does not change anything, same `/current/current/server/openapi.html` output url.

**Comment by garz75** [comment]
> I agree that your workaround is just, well, a workaround ;-)
> 
> To further tackle this down, could you attach ~your config.toml, ~ (it's already there...) the commandline you are using to run Hugo and Hugos version number to this issue? This may be related to your `baseURL` setting is my first guess.

Here is the config files:
[config.zip](https://github.com/McShelby/hugo-theme-relearn/files/7718067/config.zip)

```console
$ hugo -d netlify/current --environment netlify
Start building sites …
hugo v0.87.0+extended darwin/amd64 BuildDate=unknown

                   | EN
-------------------+-----
  Pages            | 47
  Paginator pages  |  0
  Non-page files   | 10
  Static files     | 57
  Processed images |  0
  Aliases          |  0
  Sitemaps         |  1
  Cleaned          |  0

Total in 432 ms
```

**Comment by McShelby** [comment]
It seems to work for me if you set `relativeURLs=true`.

It doesn't work for me if I set `relativeURLs=false` - although I think it should. But that's a Hugo issue and a quick search reveals all kind of different issues with permutations of `relativeURLs`, `canonifyURLs` and multilang site configurations.

Is this a feasible workaround for your issue?

**Comment by McShelby** [comment]
I did further testing. It's a mess. I tested with the latest Hugo version 0.92.1

**tldr; it only works with `canonifyURLs = true` regardless of any other setting.**

- works with or without defaultContentLanguage
  ````
  canonifyURLs = true
  relativeURLs = false
  ````

- works with or without defaultContentLanguage
  ````
  canonifyURLs = true
  relativeURLs = true
  ````

- does not work for md image links in defaultContentLanguage pages
  ````
  canonifyURLs = false
  relativeURLs = false
  ````

- does not work for md image links in defaultContentLanguage pages
  does not work for md page links in non defaultContentLanguage pages
  ````
  canonifyURLs = false
  relativeURLs = false
  defaultContentLanguage = "en"
  ````

- does not work for css links in any pages with or without defaultContentLanguage
  does not work for md image links in any pages with or without defaultContentLanguage
  does not work for md image links in any pages with or without defaultContentLanguage
  ````
  canonifyURLs = false
  relativeURLs = true
  ````

So `canonifyURLs = true` seems to do the trick. From the Hugo documentation I would assume that in this case `relativeURLs` is completly ignored. This matches my observations.

This is pretty sad, as you are now limited to the files hosted from `baseURL` exclusivly. No freedom in scheme, host, port and also you can not using it from the file system.

**Comment by McShelby** [comment]
@garz75 I may have found a solution. Forget my previous rambling.

Simply, remove the leading `/` from your URL, like:

````
baseURL: "https://docs.mysite.com/current"
relativeURLs: false
menu:
  shortcuts:
    - name: Server
      url: server/openapi.html
````

**Comment by McShelby** [comment]
As this issue is only affected by the content in `config.toml`, I've only updated the docs and exampleSite to solve this.

**Comment by McShelby** [comment]
> Hugo will not generate valid sites if you will give a subfolder in baseURL and set relativeURLs=true.

...he said, simply forgot about that in an instant and broke the official doumentation 🤦

**Comment by garz75** [comment]
Thank you very much for the time you spent on this issue, I appreciate it. I will apply your solution tomorrow when I am at work. Thanks again!
